python/pyspark/pandas/frame.py

```diff
@@ -490,20 +491,30 @@ def _internal(self) -> InternalFrame:
         return cast(InternalFrame, self._internal_frame)  # type: ignore[has-type]
 
     def _update_internal_frame(
-        self, internal: InternalFrame, requires_same_anchor: bool = True
+        self,
+        internal: InternalFrame,
+        requires_same_anchor: bool = True,
+        force_disconnect: bool = False,
     ) -> None:
         """
         Update InternalFrame with the given one.
 
-        If the column_label is changed or the new InternalFrame is not the same `anchor`,
-        disconnect the link to the Series and create a new one.
+        If the column_label is changed or the new InternalFrame is not the same `anchor` or the
+        `force_disconnect` flag is set to True, disconnect the link to the Series and create a new
+        one.
 
         If `requires_same_anchor` is `False`, checking whether or not the same anchor is ignored
         and force to update the InternalFrame, e.g., replacing the internal with the resolved_copy,
         updating the underlying Spark DataFrame which need to combine a different Spark DataFrame.
 
-        :param internal: the new InternalFrame
-        :param requires_same_anchor: whether checking the same anchor
+        Parameters
+        ----------
+        internal : InternalFrame
+            The new InternalFrame
+        requires_same_anchor : bool
+            Whether checking the same anchor
+        force_disconnect : bool
+            Force to disconnect the link to Series and create new one
```

Comment:
What do you think we call out `force_disconnect` takes effects on anchor frames only, or rename it with `anchor` included? My concern is that the word `Series` here may seem fuzzy.

Reply:
> What do you think we call out `force_disconnect` takes effects on anchor frames only, or rename it with `anchor` included? My concern is that the word `Series` here may seem fuzzy.

+1

Reply:
```suggestion
        anchor_force_disconnect : bool
            Force to disconnect the original anchor and create new one
```

How about? : )

Reply:
updated. Pls let me know if you have better idea.